### PR TITLE
feat: Add integration tests for dbt import

### PR DESCRIPTION
**Pull request by YassinNouh21**
## Summary
- Add comprehensive integration tests for dbt import functionality
- Fix bugs in test assertions (entity.join_key, object count, entity comparison)
- Add test coverage for error handling, type mapping edge cases, and code generation validation

## Test Coverage
- **37 tests** covering:
  - Manifest parsing and filtering
  - All data source types (BigQuery, Snowflake, File)
  - Type mapping including Array, Bool, parameterized types
  - Error handling (missing manifest, invalid JSON)
  - Code generation with ast.parse() validation

## Test plan
- [x] All 37 tests pass locally
- [x] Tests are isolated with pytest.ini to avoid loading heavy dependencies

---

<a href="https://app.devin.ai/review/feast-dev/feast/pull/5899" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: .github/workflows/dbt-integration-tests.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        run: |
+          uv pip install --system dbt-core dbt-duckdb
+      
+      - name: Run dbt commands
+        run: |
+          cd sdk/python/tests/integration/dbt/test_dbt_project
+          dbt deps
+          dbt build
+      
+      - name: Setup Feast project for dbt import test
```
Comment (franciscojavierarceo) [inline .github/workflows/dbt-integration-tests.yml]: Nice! Can you instead put most of the logic into the `Makefile` so that we can easily run `make dbt-integration-test` and swap that here?

File: sdk/python/pytest.ini [hunk 2]
```diff
@@ -11,7 +11,6 @@ env =
 
 filterwarnings =
     error::_pytest.warning_types.PytestConfigWarning
-    error::_pytest.warning_types.PytestUnhandledCoroutineWarning
```
Comment (franciscojavierarceo) [inline sdk/python/pytest.ini:14]: why remove this?

**Comment by franciscojavierarceo** [comment]
@YassinNouh21 can you post a screen shot of the feast UI and the dbt UI lineage graphs to see how they render? 

Also looks like you may have this hooked up to mcp to handle this, which is pretty cool

**Comment by franciscojavierarceo** [comment]
@YassinNouh21 you can spin up the UI locally with `feast ui` after you compile the dbt models.  you should also be able to render the dbt ui locally.

**Comment by YassinNouh21** [comment]
image

image

**Comment by franciscojavierarceo** [comment]
hmm the protobuf issue sounds like a real concern, no?

**Comment by YassinNouh21** [comment]
DCO check failing due to unsigned commits. Suggest squash merge with Signed-off-by to resolve.
@franciscojavierarceo

**Review by franciscojavierarceo — APPROVED** [review]

**Comment by franciscojavierarceo** [comment]
@YassinNouh21 you can also click the DCO button in the UI

**Comment by YassinNouh21** [comment]
@franciscojavierarceo PR is now clean with all tests passing. Ready for review.

File: .github/workflows/dbt-integration-tests.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  pull_request:
+    paths:
+      - 'sdk/python/feast/dbt/**'
+      - 'sdk/python/tests/integration/dbt/**'
+      - 'sdk/python/tests/unit/dbt/**'
+      - '.github/workflows/dbt-integration-tests.yml'
+
+jobs:
+  dbt-integration-test:
+    runs-on: ubuntu-latest
```
Comment (ntkathole) [inline .github/workflows/dbt-integration-tests.yml:22]: can we have a check for ok-to-test label so that resources don't get spin up for every PR raised, similar to other integration tests pattern
https://github.com/feast-dev/feast/blob/master/.github/workflows/pr_local_integration_tests.yml#L13

**Comment by YassinNouh21** [comment]
@ntkathole Done! Added the ok-to-test label check following the same pattern as pr_local_integration_tests.yml
